pwnlib/libcdb.py

```diff
@@ -255,6 +257,198 @@ def unstrip_libc(filename):
+    them in a local directory. The libraries are looked up using `libc.rip <https://libc.rip>`_
+    and fetched from the official package repositories if available.
+
+    This commonly includes the ``ld-linux-x86-64.so.2`` and ``libpthread.so.0`` binaries
+    which can be used to execute the program locally when the given libc is
+    incompatible with the local dynamic loader.
+
+    Note: Only .deb and .pkg.tar.* packages are currently supported (Debian/Ubuntu, Arch).
+
+    Arguments:
+        libc_path(str):
+            The path the libc binary.
+        unstrip(bool):
+            Try to fetch debug info for the libc and apply it to the downloaded file.
+
+    Returns:
+        The path to the cached directory containing the downloaded libraries.
+
+    Example:
+        >>> libc_path = ELF(which('ls'), checksec=False).libc.path
+        >>> lib_path = download_libraries(libc_path)
+        >>> lib_path is not None
+        True
+        >>> os.path.exists(os.path.join(lib_path, 'libc.so.6'))
+        True
+        >>> os.path.exists(os.path.join(lib_path, 'ld-linux-x86-64.so.2'))
+        True
+    """
+
+    libc = ELF(libc_path, checksec=False)
+    if not libc.buildid:
+        log.warn_once('Given libc does not have a buildid.')
+        return None
+    
+    # Handle caching and don't redownload if it already exists.
+    cache_dir = os.path.join(context.cache_dir, 'libcdb_libs')
+    if not os.path.isdir(cache_dir):
+        os.makedirs(cache_dir)
+    
+    cache_dir = os.path.join(cache_dir, enhex(libc.buildid))
+    if os.path.exists(cache_dir):
+        return cache_dir
+
+    libc_match = query_libc_rip({'buildid': enhex(libc.buildid)})
+    if libc_match is None or len(libc_match) == 0:
+        log.warn_once('Failed to find matching libraries for provided libc.')
+        return None
+
+    package_url = libc_match[0]['libs_url']
+    extension_handlers = {
+        '.deb': _extract_debfile,
+        '.pkg.tar.xz': _extract_pkgfile,
+        '.pkg.tar.zst': _extract_pkgfile,
+    }
+
+    package_filename = os.path.basename(package_url)
+    for extension, handler in extension_handlers.items():
+        if package_filename.endswith(extension):
+            break
+    else:
+        log.failure('Cannot handle %s (%s)', package_filename, package_url)
+        return None
+
+    # Download the package
+    package = wget(package_url, timeout=20)
+    if not package:
+        return None
+
+    # Create target cache directory to extract files into
+    if not os.path.isdir(cache_dir):
+        os.makedirs(cache_dir)
+
+    try:
+        # Extract the archive
+        libc_path = handler(cache_dir, package_filename, package)
+    except Exception as e:
+        os.removedirs(cache_dir)
+        log.failure('Failed to extract %s: %s', package_filename, e)
+        return None
+    # Unstrip the libc binary
+    try:
+        if unstrip:
+            unstrip_libc(libc_path)
+    except:
```

Comment:
We probably don't want to catch `KeyboardInterrupt` and friends
```suggestion
    except Exception:
```